src/GafferTestModule/ProcessTest.cpp

```diff
@@ -0,0 +1,199 @@
+// create a variety of process graphs very explicitly from ProcessTestCase.
+class TestProcess : public Process
+{
+
+	public :
+
+		TestProcess( const Plug *plug, int result, const Dependencies::ConstPtr &dependencies )
+			:	Process( g_staticType, plug, plug ), m_result( result ), m_dependencies( dependencies )
+		{
+		}
+
+		using ResultType = int;
+
+		ResultType run() const
+		{
+			const ThreadState &threadState = ThreadState::current();
+			tbb::task_group_context taskGroupContext( tbb::task_group_context::isolated );
+
+			tbb::parallel_for_each(
+
+				m_dependencies->map.begin(), m_dependencies->map.end(),
+
+				[&] ( const Dependencies::Map::value_type &dependency ) {
+
+					ThreadState::Scope scope( threadState );
+					const int expectedResult = dependency.first;
+
+					const Plug *p = plug();
+					if( const Plug *input = p->getInput() )
+					{
+						// Compute the dependencies using the plug's input if it
+						// has one, otherwise using this plug. The only reason
+						// for using an input is get more fine-grained information
+						// from the Monitors used in the unit tests (because they
+						// capture statistics per plug).
+						p = input;
+					}
+
+					int actualResult;
+					if( expectedResult >= 0 )
+					{
+						actualResult = Process::acquireCollaborativeResult<TestProcess>( expectedResult, p, expectedResult, dependency.second );
+					}
+					else
+					{
+						actualResult = TestProcess( p, expectedResult, dependency.second ).run();
+					}
+					GAFFERTEST_ASSERTEQUAL( actualResult, expectedResult );
+
+				},
+
+				taskGroupContext
+
+			);
+
+			return m_result;
+		}
+
+		using CacheType = IECorePreview::LRUCache<int, int, IECorePreview::LRUCachePolicy::Parallel>;
+		static CacheType g_cache;
+
+		static size_t cacheCostFunction( int value )
+		{
+			return 1;
+		}
+
+	private :
+
+		const int m_result;
+		const Dependencies::ConstPtr m_dependencies;
+
+		static const IECore::InternedString g_staticType;
+
+};
+
+TestProcess::CacheType TestProcess::g_cache( TestProcess::CacheType::GetterFunction(), 100000 );
+// Spoof type so that we can use PerformanceMonitor to check we get the processes we expect in ProcessTest.py.
+const IECore::InternedString TestProcess::g_staticType( "computeNode:compute" );
+
+Dependencies::ConstPtr dependenciesFromDict( dict dependenciesDict, std::unordered_map<const PyObject *, Dependencies::ConstPtr> &converted )
+{
+	auto it = converted.find( dependenciesDict.ptr() );
+	if( it != converted.end() )
+	{
+		return it->second;
+	}
+
+	Dependencies::Ptr result = new Dependencies;
+	auto items = dependenciesDict.items();
+	for( size_t i = 0, l = len( items ); i < l; ++i )
+	{
+		int v = extract<int>( items[i][0] );
+		dict d = extract<dict>( items[i][1] );
+		result->map[v] = dependenciesFromDict( d, converted );
+	}
+
+	converted[dependenciesDict.ptr()] = result;
+	return result;
+}
+
+void runTestProcess( const Plug *plug, int expectedResult, dict dependenciesDict )
```

Comment:
Do you have any suggestions for an improvement? It definitely does run a `TestProcess`, so I guess the problem is that it could be running some non-specific "test process". Would the solution be to rename `TestProcess` to something else? We do have a load of `TestFoo` classes which are instances of `Foo` designed only for testing `Foos`.

Reply:
I guess the reason it feels odd to me, is that even aside from "Process" being a generic name, this feels like it's for testing a specific aspect of Processes, and it's running a network of Processes, not a single one.

Something like runTestDependencyProcesses or something would feel clearer to me, but I don't feel strongly if you're happy with this name.